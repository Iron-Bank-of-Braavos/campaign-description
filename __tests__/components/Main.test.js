import React from 'react';
import Enzyme from 'enzyme';
import Main from '../../client/components/Main.jsx';

describe('Main Component', () => {
  it('should be defined', () => {
    expect(Main).toBeDefined();
  });

  it('should render component without props', () => {
    const mainComponent = Enzyme.shallow(<Main />);
    expect(mainComponent).toMatchSnapshot();
  })
});