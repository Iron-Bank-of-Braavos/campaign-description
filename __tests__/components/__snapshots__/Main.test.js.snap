// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component should render component without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div
        className="pad-2"
      >
        <styled.div
          className="row"
        >
          <styled.div
            className="pad-3"
          >
            <styled.h3>
              About
            </styled.h3>
            <About />
            <styled.h3>
              Risks and challenges
            </styled.h3>
          </styled.div>
          <styled.div
            className="pad-4"
          >
            <styled.h3>
              Support
            </styled.h3>
            <List />
          </styled.div>
        </styled.div>
      </styled.div>,
      "className": "pad-1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          className="row"
        >
          <styled.div
            className="pad-3"
          >
            <styled.h3>
              About
            </styled.h3>
            <About />
            <styled.h3>
              Risks and challenges
            </styled.h3>
          </styled.div>
          <styled.div
            className="pad-4"
          >
            <styled.h3>
              Support
            </styled.h3>
            <List />
          </styled.div>
        </styled.div>,
        "className": "pad-2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              className="pad-3"
            >
              <styled.h3>
                About
              </styled.h3>
              <About />
              <styled.h3>
                Risks and challenges
              </styled.h3>
            </styled.div>,
            <styled.div
              className="pad-4"
            >
              <styled.h3>
                Support
              </styled.h3>
              <List />
            </styled.div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.h3>
                  About
                </styled.h3>,
                <About />,
                <styled.h3>
                  Risks and challenges
                </styled.h3>,
                undefined,
              ],
              "className": "pad-3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "About",
                },
                "ref": null,
                "rendered": "About",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Risks and challenges",
                },
                "ref": null,
                "rendered": "Risks and challenges",
                "type": [Function],
              },
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.h3>
                  Support
                </styled.h3>,
                <List />,
              ],
              "className": "pad-4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Support",
                },
                "ref": null,
                "rendered": "Support",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          className="pad-2"
        >
          <styled.div
            className="row"
          >
            <styled.div
              className="pad-3"
            >
              <styled.h3>
                About
              </styled.h3>
              <About />
              <styled.h3>
                Risks and challenges
              </styled.h3>
            </styled.div>
            <styled.div
              className="pad-4"
            >
              <styled.h3>
                Support
              </styled.h3>
              <List />
            </styled.div>
          </styled.div>
        </styled.div>,
        "className": "pad-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div
            className="row"
          >
            <styled.div
              className="pad-3"
            >
              <styled.h3>
                About
              </styled.h3>
              <About />
              <styled.h3>
                Risks and challenges
              </styled.h3>
            </styled.div>
            <styled.div
              className="pad-4"
            >
              <styled.h3>
                Support
              </styled.h3>
              <List />
            </styled.div>
          </styled.div>,
          "className": "pad-2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div
                className="pad-3"
              >
                <styled.h3>
                  About
                </styled.h3>
                <About />
                <styled.h3>
                  Risks and challenges
                </styled.h3>
              </styled.div>,
              <styled.div
                className="pad-4"
              >
                <styled.h3>
                  Support
                </styled.h3>
                <List />
              </styled.div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.h3>
                    About
                  </styled.h3>,
                  <About />,
                  <styled.h3>
                    Risks and challenges
                  </styled.h3>,
                  undefined,
                ],
                "className": "pad-3",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "About",
                  },
                  "ref": null,
                  "rendered": "About",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Risks and challenges",
                  },
                  "ref": null,
                  "rendered": "Risks and challenges",
                  "type": [Function],
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.h3>
                    Support
                  </styled.h3>,
                  <List />,
                ],
                "className": "pad-4",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Support",
                  },
                  "ref": null,
                  "rendered": "Support",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Main should render component without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div
        className="pad-2"
      >
        <styled.div
          className="row"
        >
          <styled.div
            className="pad-3"
          >
            <styled.h3>
              About
            </styled.h3>
            <About />
            <styled.h3>
              Risks and challenges
            </styled.h3>
          </styled.div>
          <styled.div
            className="pad-4"
          >
            <styled.h3>
              Support
            </styled.h3>
            <List />
          </styled.div>
        </styled.div>
      </styled.div>,
      "className": "pad-1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          className="row"
        >
          <styled.div
            className="pad-3"
          >
            <styled.h3>
              About
            </styled.h3>
            <About />
            <styled.h3>
              Risks and challenges
            </styled.h3>
          </styled.div>
          <styled.div
            className="pad-4"
          >
            <styled.h3>
              Support
            </styled.h3>
            <List />
          </styled.div>
        </styled.div>,
        "className": "pad-2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div
              className="pad-3"
            >
              <styled.h3>
                About
              </styled.h3>
              <About />
              <styled.h3>
                Risks and challenges
              </styled.h3>
            </styled.div>,
            <styled.div
              className="pad-4"
            >
              <styled.h3>
                Support
              </styled.h3>
              <List />
            </styled.div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.h3>
                  About
                </styled.h3>,
                <About />,
                <styled.h3>
                  Risks and challenges
                </styled.h3>,
                undefined,
              ],
              "className": "pad-3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "About",
                },
                "ref": null,
                "rendered": "About",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Risks and challenges",
                },
                "ref": null,
                "rendered": "Risks and challenges",
                "type": [Function],
              },
              undefined,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.h3>
                  Support
                </styled.h3>,
                <List />,
              ],
              "className": "pad-4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Support",
                },
                "ref": null,
                "rendered": "Support",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          className="pad-2"
        >
          <styled.div
            className="row"
          >
            <styled.div
              className="pad-3"
            >
              <styled.h3>
                About
              </styled.h3>
              <About />
              <styled.h3>
                Risks and challenges
              </styled.h3>
            </styled.div>
            <styled.div
              className="pad-4"
            >
              <styled.h3>
                Support
              </styled.h3>
              <List />
            </styled.div>
          </styled.div>
        </styled.div>,
        "className": "pad-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <styled.div
            className="row"
          >
            <styled.div
              className="pad-3"
            >
              <styled.h3>
                About
              </styled.h3>
              <About />
              <styled.h3>
                Risks and challenges
              </styled.h3>
            </styled.div>
            <styled.div
              className="pad-4"
            >
              <styled.h3>
                Support
              </styled.h3>
              <List />
            </styled.div>
          </styled.div>,
          "className": "pad-2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div
                className="pad-3"
              >
                <styled.h3>
                  About
                </styled.h3>
                <About />
                <styled.h3>
                  Risks and challenges
                </styled.h3>
              </styled.div>,
              <styled.div
                className="pad-4"
              >
                <styled.h3>
                  Support
                </styled.h3>
                <List />
              </styled.div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.h3>
                    About
                  </styled.h3>,
                  <About />,
                  <styled.h3>
                    Risks and challenges
                  </styled.h3>,
                  undefined,
                ],
                "className": "pad-3",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "About",
                  },
                  "ref": null,
                  "rendered": "About",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Risks and challenges",
                  },
                  "ref": null,
                  "rendered": "Risks and challenges",
                  "type": [Function],
                },
                undefined,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.h3>
                    Support
                  </styled.h3>,
                  <List />,
                ],
                "className": "pad-4",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Support",
                  },
                  "ref": null,
                  "rendered": "Support",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
